command/setup_consul.go

```diff
@@ -0,0 +1,736 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package command
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+// Ensure SetupConsulCommand satisfies the cli.Command interface.
+var _ cli.Command = &SetupConsulCommand{}
+
+//go:embed asset/consul-wi-default-auth-method-config.json
+var consulAuthConfigBody []byte
+
+//go:embed asset/consul-wi-default-policy.hcl
+var consulPolicyBody []byte
+
+const (
+	consulAuthMethodServicesName = "nomad-services"
+	consulAuthMethodServicesDesc = "Login method for Nomad services using workload identities"
+	consulAuthMethodTasksName    = "nomad-tasks"
+	consulAuthMethodTaskDesc     = "Login method for Nomad tasks using workload identities"
+	consulRoleTasks              = "role-nomad-tasks"
+	consulPolicyName             = "policy-nomad-tasks"
+	consulNamespace              = "nomad-workloads"
+	consulAud                    = "consul.io"
+)
+
+type SetupConsulCommand struct {
+	Meta
+
+	// client is the Consul API client shared by all functions in the command
+	// to reuse the same connection.
+	client    *api.Client
+	clientCfg *api.Config
+
+	jwksURL string
+
+	consulEnt bool
+	autoYes   bool
+
+	// store IDs of objects we may need to cleanup
+	policyID       string
```

Comment:
```suggestion
```

I think we don't need this anymore. I avoid using the ID because this doesn't persist between runs, so the policy was not being cleaned up.

Reply:
fixed in [d04a235](https://github.com/hashicorp/nomad/pull/18820/commits/d04a235d90260a8dbac816af91512d5e5a20d93f)